### PR TITLE
Allow placement of blocks overriding tallgrass and fire.

### PR DESCRIPTION
**Pull request by justcool393**
Currently, players are unable to place inside the fire block, as you can in vanilla. This PR adds this ability.

Also, when a block is attempted to be replacing a tallgrass block, it creates the block on what the block was placed against.

### CODE REVIEW

File: src/main/java/net/glowstone/block/blocktype/BlockType.java [hunk 1]
```diff
@@ -117,10 +117,8 @@ public final void rightClickBlock(GlowPlayer player, GlowBlock against, BlockFac
         GlowBlock target = against.getRelative(face);
         GlowBlockState newState = target.getState();
 
-        // only allow placement inside tall-grass, air, or liquid
-        if (against.getType() == Material.LONG_GRASS) {
-            target = against;
-        } else if (!target.isEmpty() && !target.isLiquid()) {
+        // only allow placement inside tall-grass, air, liquids or fire
+        if (!target.isEmpty() && !target.isLiquid() && !(target.getType() == Material.FIRE || target.getType() == Material.LONG_GRASS)) {
```
Comment (ewized) [inline src/main/java/net/glowstone/block/blocktype/BlockType.java:121]: I don't think this is the most efficient way of handing it.

**Comment by SpaceManiac** [comment]
The change does seem to allow placement inside fire, but it also breaks placing blocks inside of tall grass by right-clicking directly on it. It's possible it didn't work before but the diff makes it look like you removed the check and it definitely doesn't work with this change.

It'd be great if you could get placement inside tall grass working again before I merged this.

**Comment by kitskub** [comment]
@SpaceManiac the check is still there...

**Comment by SpaceManiac** [comment]
When placing against tall grass, the server has to compensate by placing the block inside the block the client says it clicked against (the tall grass). That's what the removed "target = against" assignment is supposed to do. Honestly I think it was broken before but it'd be great to see it fixed here.

**Comment by kitskub** [comment]
@SpaceManiac oh I see

**Comment by justcool393** [comment]
Right now, trying to place over tall grass isn't possible to do at all. I'll look and see what I can do to fully fix it (it doesn't like me placing over it when I have it aimed at the tallgrass block.

**Comment by turt2live** [comment]
what about snow layers?

**Comment by justcool393** [comment]
Yeah, that will need to be implemented as well. Looks like it only replaces the block if the damage value is 0.

File: src/main/java/net/glowstone/block/blocktype/BlockType.java (hunk 1 above)
Comment (turt2live) [inline src/main/java/net/glowstone/block/blocktype/BlockType.java:121]: Sounds like a possible method in `Material` should be added.

**Comment by turt2live** [comment]
@justcool393 this PR does not currently merge. Please fix that ASAP!

**Comment by TomCaserta** [comment]
This is already fixed in the latest version in any case (including in fire). Overriding canOverride(block,face,holding) and canAbsorb(block,face,holding) methods in the BlockType file achieves the same results.

**Comment by SpaceManiac** [comment]
A fix for this same issue was added in 19c7ec353a9a1c09b70e8ade5e8659c60ab13bfc. Thanks for the help!

**Pull request closed without merging**